akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala

```diff
@@ -0,0 +1,151 @@
+        case GetValue(replyTo) => Effect.reply(replyTo)(state)
+      }
+    //#command-handler
+
+    //#behavior
+    def counter(id: String): DurableStateBehavior[Command[_], State] = {
+      DurableStateBehavior.apply[Command[_], State](
+        persistenceId = PersistenceId.ofUniqueId(id),
+        emptyState = State(0),
+        commandHandler = commandHandler)
+    }
+    //#behavior
+  }
+
+  //#structure
+  object MyPersistentCounter {
+    sealed trait Command[ReplyMessage] extends CborSerializable
+
+    final case class State(value: Int) extends CborSerializable
+
+    def counter(persistenceId: PersistenceId): DurableStateBehavior[Command[_], State] = {
+      DurableStateBehavior.apply[Command[_], State](
+        persistenceId,
+        emptyState = State(0),
+        commandHandler =
+          (state, command) => throw new NotImplementedError("TODO: process the command & return an Effect"))
+    }
+  }
+  //#structure
+
+  import MyPersistentCounter._
+
+  object MyPersistentCounterWithReplies {
+
+    //#effects
+    sealed trait Command[ReplyMessage] extends CborSerializable
+    final case class IncrementWithConfirmation(replyTo: ActorRef[Done]) extends Command[Done]
+    final case class GetValue(replyTo: ActorRef[State]) extends Command[State]
+
+    final case class State(value: Int) extends CborSerializable
+
+    def counter(persistenceId: PersistenceId): DurableStateBehavior[Command[_], State] = {
+      DurableStateBehavior.withEnforcedReplies[Command[_], State](
+        persistenceId,
+        emptyState = State(0),
+        commandHandler = (state, command) =>
+          command match {
+
+            case IncrementWithConfirmation(replyTo) =>
+              Effect.persist(state.copy(value = state.value + 1)).thenReply(replyTo)(_ => Done)
+
+            case GetValue(replyTo) =>
+              Effect.reply(replyTo)(state)
+          })
+    }
+    //#effects
+  }
+
+  object BehaviorWithContext {
+    // #actor-context
+    import akka.persistence.typed.state.scaladsl.Effect
+    import akka.persistence.typed.state.scaladsl.DurableStateBehavior.CommandHandler
+
+    def apply(): Behavior[String] =
+      Behaviors.setup { context =>
+        DurableStateBehavior[String, State](
+          persistenceId = PersistenceId.ofUniqueId("myPersistenceId"),
+          emptyState = State(0),
+          commandHandler = CommandHandler.command { cmd =>
+            context.log.info("Got command {}", cmd)
+            Effect.none
+          })
+      }
+    // #actor-context
+  }
+
+  object TaggingBehavior {
+    def apply(): Behavior[Command[_]] =
+      //#tagging
+      DurableStateBehavior[Command[_], State](
+        persistenceId = PersistenceId.ofUniqueId("abc"),
+        emptyState = State(0),
+        commandHandler = (state, cmd) => throw new NotImplementedError("TODO: process the command & return an Effect"))
+        .withTag("tag1")
+        .withTag("tag2")
```

Comment:
We don't support setting multiple tags (yet?), so perhaps we shouldn't give that impression in the docs ;)

Reply:
done ..